src/components/toolbar/helpers.ts

```diff
@@ -312,36 +316,70 @@ export const indentSection =
     const parentSection = $from.node(parentSectionDepth)
     const startIndex = $from.index(parentSectionDepth)
 
-    const previousSection =
-      startIndex > 0 ? parentSection.child(startIndex - 1) : null
-    const isValidContainer = previousSection?.type === nodes.section
+    // Check if there's a valid previous section to indent into
+    // Look backwards through siblings to find a valid (non-deleted, non-moved) section
+    let previousSection: Node | null = null
+    for (let i = startIndex - 1; i >= 0; i--) {
+      const candidate = parentSection.child(i)
+      if (candidate.type === nodes.section && !shouldSkipNode(candidate)) {
+        previousSection = candidate
+        break
+      }
+    }
 
     let anchor
-    if (!previousSection || !isValidContainer) {
+    if (!previousSection) {
       // No valid previous section, creating new parent section
       const emptyTitle = nodes.section_title.create()
       const newParentSectionContent = Fragment.fromArray([emptyTitle, section])
       const newParentSection = nodes.section.create({}, newParentSectionContent)
 
-      tr.replaceWith(beforeSection, afterSection, newParentSection)
+      tr.insert(beforeSection, newParentSection)
+
+      // Delete the original section (now at shifted position)
+      const newBeforeSection = beforeSection + newParentSection.nodeSize
+      const newAfterSection = afterSection + newParentSection.nodeSize
+
+      tr.delete(newBeforeSection, newAfterSection)
 
       anchor = beforeSection + 1
     } else {
       // Moving section into previous section as subsection
-      const newPreviousSection = previousSection.copy(
-        previousSection.content.append(Fragment.from(section))
-      )
+      // Find the actual position of the previous section by walking through the document
+      let beforePreviousSection: number | null = null
 
-      const beforePreviousSection = beforeSection - previousSection.nodeSize
+      $from.doc.descendants((node, pos) => {
+        if (node === previousSection) {
+          beforePreviousSection = pos
+          return false // stop iteration
+        }
+      })
 
-      tr.replaceWith(beforePreviousSection, afterSection, newPreviousSection)
+      if (beforePreviousSection === null) {
+        return false
+      }
+
+      const insertPos = beforePreviousSection + previousSection.nodeSize - 1
+      tr.insert(insertPos, section)
+
+      // Delete the original section (positions have shifted due to insert)
+      const newBeforeSection = beforeSection + section.nodeSize
+      const newAfterSection = afterSection + section.nodeSize
+      tr.delete(newBeforeSection, newAfterSection)
 
-      anchor = beforePreviousSection + 1
+      anchor = insertPos + 1
     }
 
     tr.setSelection(TextSelection.create(tr.doc, anchor))
 
-    dispatch(skipTracking(tr))
+    // Set indentation metadata for track changes
+    tr.setMeta('indentation', {
+      type: 'indent',
```

Comment:
it seems like NodeType can be found out of the context of the change itself. And I'm really not sure if we need to consider container creation a part of this change, because the container can be later used to insert more content and rejecting it all as indentation will be problematic. (I also made a related comment about that in TC plugin PR)